Document unhealthy status (#12125)
kyma-project/kyma: docs/05-technical-reference/svls-02-function-processing-stages.md

Thanks to the implemented reconciliation loop, the Function Controller constantly observes all newly created or updated resources. If it detects changes, it fetches the appropriate resource's status and only then updates the Function's status.

> Function Controller observes the status of the underlying deployment. In case the minimum availibility condition of the replicas is not satisfied, the Function Controller sets the `Running` status to `Unknown` with reason `MinimumReplicasUnavailable`. Such Function should be considered unhealthy and the runtime profile or amount of replicas should be adjusted.
> 

Reviewer: ```suggestion
The Function Controller observes the status of the underlying deployment. If the minimum availability condition for the replicas is not satisfied, the Function Controller sets the **Running** status to `Unknown` with reason `MinimumReplicasUnavailable`. Such a Function should be considered unhealthy and the runtime profile or number of replicas must be adjusted.

```
- `In case` is not used like this (but people often do this). Please use `if` whenever possible. 
- Replicas are countable, so it's `number of`, not `amount of`. 
- Do I understand correctly that there's a status called **Running** which can be set to `true`, `false`, or `unknown`? Or is it **Status** whose values are `running` or `unknown`? (I'm asking here what's the parameter/or column name, to simplify it/, and what's the value.)

> The Function Controller observes the status of the underlying deployment. If the minimum availability condition for the replicas is not satisfied, the Function Controller sets the **Running** status to `Unknown` with reason `MinimumReplicasUnavailable`. Such a Function should be considered unhealthy and the runtime profile or number of replicas must be adjusted.

Reviewer: And as per [our guidelines](https://github.com/kyma-project/community/blob/main/guidelines/content-guidelines/08-style-and-terminology.md#capitalized-terms), we capitalize the names of Kubernetes resources. 
```suggestion
The Function Controller observes the status of the underlying Deployment. If the minimum availability condition for the replicas is not satisfied, the Function Controller sets the **Running** status to `Unknown` with reason `MinimumReplicasUnavailable`. Such a Function should be considered unhealthy and the runtime profile or number of replicas must be adjusted.
```
